atomic_reactor/plugins/build_imagebuilder.py

```diff
@@ -0,0 +1,70 @@
+"""
+Copyright (c) 2017 Red Hat, Inc
+All rights reserved.
+
+This software may be modified and distributed under the terms
+of the BSD license. See the LICENSE file for details.
+"""
+from __future__ import print_function, unicode_literals
+
+import subprocess
+from time import sleep
+
+from atomic_reactor.plugin import BuildStepPlugin
+from atomic_reactor.build import BuildResult
+from atomic_reactor.constants import CONTAINER_IMAGEBUILDER_BUILD_METHOD
+
+
+class ImagebuilderPlugin(BuildStepPlugin):
+    """
+    Build image using imagebuilder https://github.com/openshift/imagebuilder
+    This requires the imagebuilder executable binary to be in $PATH.
+    """
+
+    key = CONTAINER_IMAGEBUILDER_BUILD_METHOD
+
+    def run(self):
+        """
+        Build image inside current environment using imagebuilder;
+        It's expected this may run within (privileged) docker container.
+
+        Returns:
+            BuildResult
+        """
+        builder = self.workflow.builder
+
+        image = builder.image.to_str()
+        # TODO: directly invoke go imagebuilder library in shared object via python module
+        ib_process = subprocess.Popen([
+            'imagebuilder', '-t', image, builder.df_dir,
+        ], stdout=subprocess.PIPE, stderr=subprocess.PIPE)
+
+        self.log.debug('build is submitted, waiting for it to finish')
+        (output, errors) = ([], [])
+        while True:
+            poll = ib_process.poll()
+            out = ib_process.stdout.readline().decode()
```

Comment:
python3 tests fail because `decode()` no longer applies to strings. I'm at a loss for how get this to pass for both python 2 and 3. I added the `decode` due to this comment [here](https://www.endpoint.com/blog/2015/01/28/getting-realtime-output-using-python):

> In Python 3.x the process will hang because 'output' is a byte array instead of a string.
> Make sure you decode readline() into a string thus:
>
> output = process.stdout.readline().decode()

This is fine for 2 but not 3. And the other suggestion, adding an encoding:

> process = subprocess.Popen(cmd, stdout=subprocess.PIPE, encoding='utf8')

Works for 3 but not 2. Suggestions please... Simplest solution might be to use `communicate()` and just not try to retrieve the logs until it completes, but that feels wrong too.

Reply:
you can do the same as docker does

if six.PY3:
   data = data.decode('utf-8')

Reply:
Thanks for that, it was simpler than I was thinking :)